### PR TITLE
Early-return NullSafeMethodCall in NodeNameResolver

### PR DESCRIPTION
**Pull request by staabm**
Prevent unnecessary work for `NullSafeMethodCall`. Instead handle it like other MethodCall's and early return.

### CODE REVIEW

**Comment by samsonasik** [comment]
is current behavior return false as well, but require loop?

**Comment by staabm** [comment]
I don't get what you mean..? Which line are you referring to? Which loop?

**Comment by samsonasik** [comment]
line 89

**Comment by samsonasik** [comment]
How about move check to isSingleName method?

File: src/NodeNameResolver/NodeNameResolver.php [hunk 1]
```diff
@@ -71,11 +73,7 @@ public function isNames(Node $node, array $names): bool
      */
     public function isName(Node | array $node, string $name): bool
     {
-        if ($node instanceof MethodCall) {
-            return false;
-        }
-
-        if ($node instanceof StaticCall) {
+        if ($node instanceof CallLike) {
```
Comment (samsonasik) [inline src/NodeNameResolver/NodeNameResolver.php]: `FuncCall` should be allowed to get name

Comment (samsonasik) [reply src/NodeNameResolver/NodeNameResolver.php]: actually, the check can be moved to `isSingleName()` as the node can be array, so only single location for check

File: src/NodeNameResolver/NodeNameResolver.php [hunk 2]
```diff
@@ -100,11 +95,7 @@ public function isCaseSensitiveName(Node $node, string $name): bool
             return false;
         }
 
-        if ($node instanceof MethodCall) {
-            return false;
-        }
-
-        if ($node instanceof StaticCall) {
+        if ($node instanceof CallLike && !$node instanceof FuncCall) {
```
Comment (staabm) [inline src/NodeNameResolver/NodeNameResolver.php:98]: I wonder about `New_`. was it also missing from this condition or intentionally not if-cased?

File: src/NodeNameResolver/NodeNameResolver.php (hunk 1 above)
Comment (staabm) [reply src/NodeNameResolver/NodeNameResolver.php]: good catch.

File: src/NodeNameResolver/NodeNameResolver.php (hunk 2 above)
Comment (samsonasik) [reply src/NodeNameResolver/NodeNameResolver.php:98]: is passing `New_` previously got the `true` or  same `false` ? that same question with `NullsafeMethodCall`

Comment (staabm) [reply src/NodeNameResolver/NodeNameResolver.php:98]: my testing suggests it is now the same as before. the test-suite is still green, which matches my local observation.

the `New_` class also does not define a name property, so its "unnamed".

Comment (samsonasik) [reply src/NodeNameResolver/NodeNameResolver.php:98]: `New_` is already instanceof `CallLike`

**Comment by samsonasik** [comment]
I think PR title should be something like "Early check NullSafeMethodCall to return false on NodeNameResolver" to avoid invalid assumption.

**Comment by staabm** [comment]
agree 👍

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Let's give it a try, thank you @staabm

**Pull request merged**